pkg/client/client.go

```diff
@@ -76,6 +79,7 @@ type DataSetClient struct {
 	addEventsMutex    sync.Mutex
 	addEventsPubSub   *pubsub.PubSub
 	addEventsChannels map[string]chan interface{}
+	userAgent         string
 }
 
 func NewClient(cfg *config.DataSetConfig, client *http.Client, logger *zap.Logger) (*DataSetClient, error) {
```

Comment:
I suppose we need mechanism how to pass `OTel collector version` to client, I would suggest to use builder pattern here and pass otel collector version using one of the builder methods.